lib/src/charts/data/core/cartesian/cartesian_data.dart

```diff
@@ -23,28 +20,61 @@ enum CartesianChartAlignment {
 enum CartesianChartOrientation { vertical, horizontal }
 
 abstract class CartesianSeries {
-  T when<T>({required T Function() barSeries}) {
+  T when<T>({
+    required T Function(BarSeries series) onBarSeries,
+  }) {
+    switch (runtimeType) {
+      case BarSeries:
+        return onBarSeries(this as BarSeries);
+      default:
+        throw TypeError();
+    }
+  }
+
+  T maybeWhen<T>({
+    T Function(BarSeries series)? onBarSeries,
+    required T Function() orElse,
+  }) {
     switch (runtimeType) {
       case BarSeries:
-        return barSeries();
+        return onBarSeries?.call(this as BarSeries) ?? orElse();
+      default:
+        throw TypeError();
+    }
+  }
+
+  static T whenSeries<T>(
```

Comment:
What's the difference in this and other when statements? It's not clear from the name. 
Maybe rename this to `whenSeriesType`

Reply:
renamed to `whenType`. Require static method when you do not have instance but know the type of Series, you are looking for